conn.go

```diff
@@ -1020,6 +1020,37 @@ func (c *Conn) ReadPartitions(topics ...string) (partitions []Partition, err err
 	return
 }
 
+// Lists all topics matching specified regex
+func (c *Conn) ListTopics() (topics []string, err error) {
```

Comment:
Could we list topics through `ReadPartitions` instead? Something similar to what's in here https://github.com/segmentio/kafka-go/pull/521

```
partitions, err := conn.ReadPartitions()
if err != nil {
    return err
}
m := map[string]struct{}{}
for _, p := range partitions {
    m[p.Topic] = struct{}{}
}
```

We could have this as a private function in topicscanner.go as well

Reply:
It seems like there is some demand for a public ListTopics function given the other PRs open to add that functionality, so perhaps we should add a public function here. I would be interested to understand the relative trade off between these differing implementations, using c.readOperation() and conn.ReadPartitions().